packages/www/components/Dashboard/AssetsTable/helpers.tsx

```diff
@@ -75,15 +75,18 @@ export const rowsPageFromState = async (
   getTasks: Function,
   onDeleteAsset: Function
 ): Promise<RowsPageFromStateResult<AssetsTableData>> => {
-  const [assets, nextCursor, count] = await getAssets(userId, {
+  const assetsPromise = getAssets(userId, {
     filters: formatFiltersForApiRequest(state.filters),
     limit: state.pageSize.toString(),
     cursor: state.cursor,
     order: state.order,
     count: true,
   });
-
-  const [tasks] = await getTasks(userId);
+  const tasksPromise = getTasks(userId, {
+    limit: state.pageSize.toString(),
+  });
+  const [assets, nextCursor, count] = await assetsPromise;
```

Comment:
Could we do a Promise.all here? I think these will still execute serially

Reply:
Yes these will still execute serially. As you suggested `... = await Promise.all([assetsPromise, tasksPromise)` will make them be fetched in parallel.

Reply:
Hmm are you sure? As far as I (thought) I knew, `await` doesn't have relation with when the awaited task executes, only when you yield the execution back to the event loop for it do execute anything that is scheduled. If it wasn't like that, how could we get those "uncaught exception" errors when no one is awaiting the promise? If the exception happens it means the code executes even with no one awaiting.

So in this case, we'd _start_ 2 jobs, enqueing them for execution, and once we `await` the first one (or anything else) the 2 jobs are free to execute and do so in parallel (not concurrently).

This behavior is also what I observed when I tested this. The 2 requests are made at the same time, it's not serial (try it out!).

Am I missing something here? I can do an isolated test when I get to the PC tomorrow,  but on these online fiddles I tried they don't let you do async stuff.

Reply:
I am pretty sure that the execution in that scope is suspended when the `await` is hit, until the promise is resolved. Then the next line is executed.

https://programmingwithswift.com/run-async-await-in-parallel-with-javascript/

Reply:
Mostly because of typing issues which will get pretty annoying with those complex return types. Each promise has its own type so the combined type will be just a union of all the types, but maybe an `as const` will help here. Worst case I can do a separate`await` all and then get each result separately.

Coming from C# it's pretty clear to me that these run in parallel, but I understand it can be confusing in the single threaded environment of JS. So I can definitely follow the most common pattern here. Maybe error handling will be better as well with that.

Reply:
Implemented it here!
https://github.com/livepeer/studio/pull/1466/commits/f26695dddfb274e894371f4aec3e05a7e77fd89f